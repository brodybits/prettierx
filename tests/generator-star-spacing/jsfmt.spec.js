run_spec(__dirname, ["babel", "babel-flow", "flow", "typescript"], {
  generatorStarSpacing: true
});
run_spec(__dirname, ["babel", "babel-flow", "flow", "typescript"], {
  generatorStarSpacing: true,
  spaceBeforeFunctionParen: true
});
