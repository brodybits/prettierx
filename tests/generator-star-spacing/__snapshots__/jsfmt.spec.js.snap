// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct.js 1`] = `
====================================options=====================================
generatorStarSpacing: true
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
// correct for:
// * "generator-star-spacing": [ "error", { "before": true, "after": true } ],
// but *not* for the following standard.js rules:
// * "space-before-function-paren": [ "error", "always" ],
// * "semi": [ "error", "never" ],

function * generator() {}
var anonymous = function * () {};
var shorthand = { * generator() {} };
class Example {
  async * foo() {}
}

=====================================output=====================================
// correct for:
// * "generator-star-spacing": [ "error", { "before": true, "after": true } ],
// but *not* for the following standard.js rules:
// * "space-before-function-paren": [ "error", "always" ],
// * "semi": [ "error", "never" ],

function * generator() {}
var anonymous = function * () {};
var shorthand = { * generator() {} };
class Example {
  async * foo() {}
}

================================================================================
`;

exports[`correct.js 2`] = `
====================================options=====================================
generatorStarSpacing: true
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
spaceBeforeFunctionParen: true
                                                                                | printWidth
=====================================input======================================
// correct for:
// * "generator-star-spacing": [ "error", { "before": true, "after": true } ],
// but *not* for the following standard.js rules:
// * "space-before-function-paren": [ "error", "always" ],
// * "semi": [ "error", "never" ],

function * generator() {}
var anonymous = function * () {};
var shorthand = { * generator() {} };
class Example {
  async * foo() {}
}

=====================================output=====================================
// correct for:
// * "generator-star-spacing": [ "error", { "before": true, "after": true } ],
// but *not* for the following standard.js rules:
// * "space-before-function-paren": [ "error", "always" ],
// * "semi": [ "error", "never" ],

function * generator () {}
var anonymous = function * () {};
var shorthand = { * generator () {} };
class Example {
  async * foo () {}
}

================================================================================
`;

exports[`incorrect.js 1`] = `
====================================options=====================================
generatorStarSpacing: true
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
// *incorrect* for the following standard.js rules:
// * "generator-star-spacing": [ "error", { "before": true, "after": true } ],
// * "space-before-function-paren": [ "error", "always" ],
// * "semi": [ "error", "never" ],

function* generator() {}
var anonymous = function* () {};
var shorthand = { * generator() {} };
function *generator() {}
var anonymous = function *() {};
var shorthand = { *generator() {} };

=====================================output=====================================
// *incorrect* for the following standard.js rules:
// * "generator-star-spacing": [ "error", { "before": true, "after": true } ],
// * "space-before-function-paren": [ "error", "always" ],
// * "semi": [ "error", "never" ],

function * generator() {}
var anonymous = function * () {};
var shorthand = { * generator() {} };
function * generator() {}
var anonymous = function * () {};
var shorthand = { * generator() {} };

================================================================================
`;

exports[`incorrect.js 2`] = `
====================================options=====================================
generatorStarSpacing: true
parsers: ["babel", "babel-flow", "flow", "typescript"]
printWidth: 80
spaceBeforeFunctionParen: true
                                                                                | printWidth
=====================================input======================================
// *incorrect* for the following standard.js rules:
// * "generator-star-spacing": [ "error", { "before": true, "after": true } ],
// * "space-before-function-paren": [ "error", "always" ],
// * "semi": [ "error", "never" ],

function* generator() {}
var anonymous = function* () {};
var shorthand = { * generator() {} };
function *generator() {}
var anonymous = function *() {};
var shorthand = { *generator() {} };

=====================================output=====================================
// *incorrect* for the following standard.js rules:
// * "generator-star-spacing": [ "error", { "before": true, "after": true } ],
// * "space-before-function-paren": [ "error", "always" ],
// * "semi": [ "error", "never" ],

function * generator () {}
var anonymous = function * () {};
var shorthand = { * generator () {} };
function * generator () {}
var anonymous = function * () {};
var shorthand = { * generator () {} };

================================================================================
`;
